sw/airborne/math/pprz_algebra_float.c

```diff
@@ -638,6 +665,38 @@ void float_eulers_of_quat(struct FloatEulers *e, struct FloatQuat *q)
   e->psi = atan2f(dcm01, dcm00);
 }
 
+/**
+ * @brief euler rotation 'YXZ'
+ * This function calculates from a quaternion the Euler angles with the order YXZ,
+ * so pitch, roll, yaw, instead of the conventional ZYX order.
+ * See https://en.wikipedia.org/wiki/Euler_angles
+ *
+ * @param e Euler output
+ * @param q Quat input
+ */
+void float_eulers_of_quat_yxz(struct FloatEulers *e, struct FloatQuat *q)
+{
+  const float qx2  = q->qx * q->qx;
+  const float qy2  = q->qy * q->qy;
+  const float qz2  = q->qz * q->qz;
+  const float qi2  = q->qi * q->qi;
+  const float qiqx = q->qi * q->qx;
+  const float qiqy = q->qi * q->qy;
+  const float qiqz = q->qi * q->qz;
+  const float qxqy = q->qx * q->qy;
+  const float qxqz = q->qx * q->qz;
+  const float qyqz = q->qy * q->qz;
+  const float r11  = 2 * (qxqz + qiqy);
+  const float r12  = qi2 - qx2 + qy2 + qz2;
+  const float r21  = -2 * (qyqz - qiqx);
+  const float r31  = 2 * (qxqy + qiqz);
```

Comment:
same here, 2.f

Reply:
Without the point, wouldn't 2 be an int that is then promoted to float?